### PR TITLE
Better fix for virus spawning, no infinite loop

### PR DESCRIPTION
**Pull request by JMcB17**
As you can see I've fixed some duplicated code by making safeSpawn a method of Server, but more importantly I've put in a simple 10-loop limit that will prevent an infinite loop if a cell fails to spawn safely.

Feel free to close if you're already working on a better solution.

### CODE REVIEW

**Comment by TopoR1** [comment]
Still not the best solution. using loops slows down performance. you'd be better off using async functions and just letting you wait instead of checking. I'm sure there is a better way to fix this.

File: src/Server.js [hunk 1]
```diff
@@ -704,9 +720,13 @@ class Server {
         }
         // Spawn player safely (do not check minions)
         var cell = new Entity.PlayerCell(this, player, pos, size);
-        if (this.willCollide(cell) && !player.isMi)
```
Comment (Tombez) [inline src/Server.js:707]: This change introduces a bug where players get one shot at a safe start position, then they start playing at that position, whether safe or not. And minions actually get spawned safely.
That is the exact opposite logic of that which was intended here.

File: src/Server.js [hunk 2]
```diff
@@ -704,9 +720,13 @@ class Server {
...
-            pos = this.randomPos(); // Not safe => retry
-        this.addNode(cell);
+        if (!player.isMi) {
+            this.addNode(cell);
+        }
+        else {
+            this.safeSpawn(cell);
+        }
+
```
Comment (Tombez) [inline src/Server.js:729]: This change introduces a bug where players get one shot at a safe start position, then they start playing at that position, whether safe or not. And minions actually get spawned safely.
That is the exact opposite logic of that which was intended here.

Comment (JMcB17) [reply src/Server.js:729]: Oops, you're right. Line 723 should be
```js
        if (player.isMi) {
```
(exclamation mark removed)

**Pull request merged**